Rakefile

```diff
@@ -174,12 +189,102 @@ desc 'update tmate'
 file 'public/files/tmate-static-linux-amd64.tar.gz' => 'public/files' do
   latest_release = latest_release_for 'tmate-io/tmate'
   logger.info "Latest tmate release is #{latest_release}"
-  fetch_githubusercontent_file "tmate-io/tmate/releases/download/#{latest_release}/tmate-#{latest_release}-static-linux-amd64.tar.gz", "github.com", 'tmate-static-linux-amd64.tar.gz'
+  fetch_githubusercontent_file(
+    File.join(
+      'tmate-io/tmate/releases/download',
+      latest_release,
+      "tmate-#{latest_release}-static-linux-amd64.tar.gz",
+    ),
+    host: "github.com", to: 'tmate-static-linux-amd64.tar.gz'
+  )
 end
 
 desc 'update rustup'
 file 'public/files/rustup-init.sh' => 'public/files' do
-  fetch_githubusercontent_file "", "sh.rustup.rs", "rustup-init.sh"
+  fetch_githubusercontent_file '', host: 'sh.rustup.rs', to: 'rustup-init.sh'
+end
+
+desc 'update raw gimme versions'
+file 'public/files/gimme-versions-binary-linux' => 'tmp' do
+  fetch_githubusercontent_file(
+    'travis-ci/gimme/master/.testdata/sample-binary-linux',
+    to: 'gimme-versions-binary-linux',
+    mode: 0o644
+  )
+end
+
+desc 'update gimme versions'
+file 'public/files/gimme-versions-binary-linux.json' => 'public/files/gimme-versions-binary-linux' do
+  raw = File.read('public/files/gimme-versions-binary-linux').split(/\n/).reject do |line|
+    line.strip.empty? || line.strip.start_with?('#')
+  end
+
+  raw.sort! do |a, b|
+    a.split('.').map(&:to_i) <=> b.split('.').map(&:to_i)
+  end
+
+  out = {}
+  raw.each do |full_version|
+    fullparts = full_version.split('.')
+    major = fullparts.first
+    out.merge!(
+      full_version => full_version,
+      "#{major}" => full_version,
+      "#{major}.x" => full_version,
+      "#{major}.x.x" => full_version,
+      "#{fullparts[0]}.#{fullparts[1]}" => full_version,
+      "#{fullparts[0]}.#{fullparts[1]}.x" => full_version
+    )
+  end
```

Comment:
Isn't this going to end up at the very end with `1.x` pointing at `1.2.2` rather than `1.8.3`?  What do we do about things like `1.9beta1`?  Just ignore them?